File: src/main/java/me/thesourcecode/sourcebot/commands/administrator/RestartCommand.java
package me.thesourcecode.sourcebot.commands.administrator;

import me.thesourcecode.sourcebot.api.Source;
import me.thesourcecode.sourcebot.api.command.Command;
import me.thesourcecode.sourcebot.api.command.CommandInfo;
import me.thesourcecode.sourcebot.api.entity.SourceRole;
import me.thesourcecode.sourcebot.api.message.alerts.CriticalAlert;
import me.thesourcecode.sourcebot.api.message.alerts.SuccessAlert;
import me.thesourcecode.sourcebot.api.utility.Utility;
import net.dv8tion.jda.api.entities.Message;
import net.dv8tion.jda.api.entities.User;

public class RestartCommand extends Command {
    private final CommandInfo INFO = new CommandInfo(
            "restart",
            "Restarts the bot.",
            CommandInfo.Category.ADMIN
    ).withControlRoles(SourceRole.ADMIN, SourceRole.OWNER);

    @Override
    public CommandInfo getInfo() {
        return INFO;
    }

    @Override
    public Message execute(Source source, Message message, String[] args) {
        User user = message.getAuthor();
        SuccessAlert sAlert = new SuccessAlert();
        sAlert.setDescription("You have successfully restarted the bot!");
        message.getChannel().sendMessage(sAlert.build(user)).queue();

        try {
            Utility.safeStop();

            String jarname = source.isBeta() ? "sourcebeta" : "sourcebot";

            Runtime.getRuntime().exec("systemctl --user restart " + jarname);
        } catch (Exception ex) {
            ex.printStackTrace();

            CriticalAlert alert = new CriticalAlert();
            alert.setTitle("ERROR").setDescription("The bot was unable to restart");
            message.getChannel().sendMessage(alert.build(user)).queue();
            return null;
        }
        return null;
    }
}

File: API/src/main/kotlin/net/sourcebot/impl/command/RestartCommand.kt
package net.sourcebot.impl.command

import net.dv8tion.jda.api.entities.Message
import net.sourcebot.api.command.RootCommand
import net.sourcebot.api.command.argument.Arguments
import net.sourcebot.api.response.EmptyResponse
import net.sourcebot.api.response.ErrorResponse
import net.sourcebot.api.response.InfoResponse
import net.sourcebot.api.response.Response

class RestartCommand(
    private val script: String
) : RootCommand() {
    override val name = "restart"
    override val description = "Restart the bot."
    override val requiresGlobal = true

    override fun execute(message: Message, args: Arguments): Response {
        message.channel.sendMessage(
            InfoResponse(
                "Restart Scheduled",
                "The bot has been scheduled to restart."
            ).asMessage(message.author)
        ).complete()
        return try {
            Runtime.getRuntime().exec(script)
            EmptyResponse()
        } catch (ex: Throwable) {
            ErrorResponse(
                "Restart Failure",
                "There was a problem restarting the bot."
            ).addField("Exception:", ex.message, false) as Response
        }
    }
}